File: YgoTFBuilder/Front-End/ygotf1builder/src/domain/Card.js
class Card {
  constructor(
    id,
    name,
    gameName,
    rarity,
    typeline,
    type,
    humanReadableCardType,
    frameType,
    desc,
    race,
    atk,
    def,
    level,
    attribute,
    archetype,
    image,
    imageCropped
  ) {
    this.id = id;
    this.name = name;
    this.gameName = gameName;
    this.rarity = rarity;
    this.typeline = typeline;
    this.type = type;
    this.humanReadableCardType = humanReadableCardType;
    this.frameType = frameType;
    this.desc = desc;
    this.race = race;
    this.atk = atk;
    this.def = def;
    this.level = level;
    this.attribute = attribute;
    this.archetype = archetype;
    this.image = image;
    this.imageCropped = imageCropped;
  }

  static buildEntity(id, data, rarity) {
    const {
      name,
      typeline,
      type,
      humanReadableCardType,
      frameType,
      desc,
      race,
      atk,
      def,
      level,
      attribute,
      archetype,
      card_images,
    } = data;

    return new Card(
      id,
      name,
      name,
      rarity,
      typeline,
      type,
      humanReadableCardType,
      frameType,
      desc,
      race,
      atk,
      def,
      level,
      attribute,
      archetype,
      card_images[0].image_url,
      card_images[0].image_url_cropped
    );
  }

  static buildEntityWithOldName(id, data, rarity, gameName) {
    const {
      name,
      typeline,
      type,
      humanReadableCardType,
      frameType,
      desc,
      race,
      atk,
      def,
      level,
      attribute,
      archetype,
      card_images,
    } = data;

    return new Card(
      id,
      name,
      gameName,
      rarity,
      typeline,
      type,
      humanReadableCardType,
      frameType,
      desc,
      race,
      atk,
      def,
      level,
      attribute,
      archetype,
      card_images[0].image_url,
      card_images[0].image_url_cropped
    );
  }

  static buildEntityWithOldNameAndNoId(data, rarity, gameName) {
    const {
      name,
      typeline,
      type,
      humanReadableCardType,
      frameType,
      desc,
      race,
      atk,
      def,
      level,
      attribute,
      archetype,
      card_images,
    } = data;

    return new Card(
      null,
      name,
      gameName,
      rarity,
      typeline,
      type,
      humanReadableCardType,
      frameType,
      desc,
      race,
      atk,
      def,
      level,
      attribute,
      archetype,
      card_images[0].image_url,
      card_images[0].image_url_cropped
    );
  }
}

export default Card;

File: YgoTFBuilder/API/YuGiOh-Builder/src/main/java/com/runage/ygobuilder/entity/Card.java
package com.runage.ygobuilder.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Objects;

@Entity
@Table(name = "card")
@Builder
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Card {

    @Id
    @Column
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column
    private String name;

    @Column(name = "game_name")
    private String gameName;

    @Column
    private String rarity;

    @Column
    private String typeline;

    @Column
    private String type;

    @Column(name = "human_readable_card_type")
    private String humanReadableCardType;

    @Column(name = "frame_type")
    private String frameType;

    @Column
    private String description;

    @Column
    private String race;

    @Column
    private Integer atk;

    @Column
    private Integer def;

    @Column
    private Short level;

    @Column
    private String attribute;

    @Column
    private String archetype;

    @Column
    private String image;

    @Column(name = "image_cropped")
    private String imageCropped;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Card card = (Card) o;
        return Objects.equals(name, card.name);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(name);
    }
}
